ooniprobe/View/TestResults/LogViewController.m

```diff
@@ -15,27 +15,49 @@ - (void)viewDidLoad {
                                             action:@selector(copy_clipboard:)];
-
-    if ([self.type isEqualToString:@"log"] || [self.type isEqualToString:@"json"]){
-        NSString *fileName;
-        if ([self.type isEqualToString:@"log"])
-            fileName = [self.measurement getLogFile];
-        else if ([self.type isEqualToString:@"json"])
-            fileName = [self.measurement getReportFile];
+    NSString *fileName;
+    if ([self.type isEqualToString:@"log"]){
+        fileName = [self.measurement getLogFile];
         NSString *content = [TestUtility getUTF8FileContent:fileName];
         if (content != nil) {
-            if ([self.type isEqualToString:@"json"]){
-                NSError *error;
-                NSData *jsonData = [content dataUsingEncoding:NSUTF8StringEncoding];
-                id jsonObject = [NSJSONSerialization JSONObjectWithData:jsonData options:0 error:&error];
-                if (error != nil) return;
-                NSData *prettyJsonData = [NSJSONSerialization dataWithJSONObject:jsonObject options:NSJSONWritingPrettyPrinted error:&error];
-                if (error != nil) return;
-                NSString *prettyPrintedJson = [NSString stringWithUTF8String:[prettyJsonData bytes]];
-                [self.textView setText:prettyPrintedJson];
-            }
-            else
-                [self.textView setText:content];
+            [self.textView setText:content];
+        }
+        else {
+            //Show Log not found alert, go back on OK.
+            //This will be useful for when we'll implement the auto log deletion
+            UIAlertAction* okButton = [UIAlertAction
+                                       actionWithTitle:NSLocalizedString(@"Modal.OK", nil)
+                                       style:UIAlertActionStyleDefault
+                                       handler:^(UIAlertAction * action) {
+                                           [self.navigationController popViewControllerAnimated:YES];
+                                       }];
+            [MessageUtility alertWithTitle:NSLocalizedString(@"Modal.Error.LogNotFound", nil)
+                                   message:nil
+                                  okButton:okButton
+                              cancelButton:nil
+                                    inView:self];
+        }
+    }
+    else if ([self.type isEqualToString:@"json"]){
+        fileName = [self.measurement getReportFile];
+        NSString *content = [TestUtility getUTF8FileContent:fileName];
+        if (content != nil) {
+            NSString *prettyPrintedJson = [self prettyPrintedJsonfromString:content];
+            [self.textView setText:prettyPrintedJson];
+        }
+        else {
+            //Download content from web
+            [self.measurement getExplorerUrl:^(NSString *measurement_url){
+                [TestUtility downloadJson:measurement_url
+                                onSuccess:^(NSDictionary *urls) {
```

Comment:
I believe the name in this variable is not correct. Is this a set of `urls`? I thought this was the content of a file.

Reply:
wrong var name (type was correct)